data_structures/Trie/Trie.py

```diff
@@ -0,0 +1,81 @@
+"""
+A Trie/Prefix Tree is a kind of search tree used to provide quick lookup
+of words/patterns in a set of words. A basic Trie however has O(n^2) space complexity
+making it impractical in practice. It however provides O(max(search_string, length of longest word)) lookup
+time making it an optimal approach when space is not an issue.
+
+This implementation assumes the character $ is not in any of the words. This character is used in the implementation
```

Comment:
Can you please find some way to achieve the desired result without setting a predefined character? An algorithm is supposed to consider all test cases in mind. Good job though!

Reply:
Thanks, no prob! I'll work on it